From 04b649f5c7a37a89b95d5ceef24fdbc0ed8c5587 Mon Sep 17 00:00:00 2001
From: Andrew Grieve <agrieve@chromium.org>
Date: Mon, 17 May 2021 19:56:06 +0000
Subject: [PATCH] Py2-Py3 fixup for dom_distiller_js

Previous attempt to make it python3-compatible had it actually still
using python2.

Bug: 1205622, 1209002
Change-Id: Ia756c178bd1ef3839496eb0d159729da7c3162ec
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2897343
Auto-Submit: Andrew Grieve <agrieve@chromium.org>
Reviewed-by: Tommy Nyquist <nyquist@chromium.org>
Commit-Queue: Andrew Grieve <agrieve@chromium.org>
Cr-Commit-Position: refs/heads/master@{#883578}
---
 .../protoc_plugins/json_values_converter.py               | 2 +-
 .../dom_distiller_js/protoc_plugins/util/plugin_protos.py | 8 ++++----
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py b/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
index 954a410275..3921a467b1 100755
--- a/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
+++ b/src/third_party/dom_distiller_js/protoc_plugins/json_values_converter.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2016 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/third_party/dom_distiller_js/protoc_plugins/util/plugin_protos.py b/src/third_party/dom_distiller_js/protoc_plugins/util/plugin_protos.py
index d3cbaf1a87..352591ce06 100644
--- a/src/third_party/dom_distiller_js/protoc_plugins/util/plugin_protos.py
+++ b/src/third_party/dom_distiller_js/protoc_plugins/util/plugin_protos.py
@@ -33,7 +33,7 @@ class PluginRequest(object):
     return dict((v.split('=') for v in self.proto.parameter.split(',')))
 
   def GetAllFiles(self):
-    files = map(ProtoFile, self.proto.proto_file)
+    files = [ProtoFile(x) for x in self.proto.proto_file]
     for f in files:
       assert f.Filename() in self.proto.file_to_generate
     return files
@@ -120,7 +120,7 @@ class ProtoFile(object):
   def GetDependencies(self):
     # import is not supported
     assert [] == self.proto.dependency
-    return map(types.GetProtoFileForFilename, self.proto.dependency)
+    return [types.GetProtoFileForFilename(x) for x in self.proto.dependency]
 
   def JavaFilename(self):
     return '/'.join(self.JavaQualifiedOuterClass().split('.')) + '.java'
@@ -164,7 +164,7 @@ class ProtoMessage(object):
     return types.TitleCase(self.proto.name)
 
   def GetFields(self):
-    return map(ProtoField, self.proto.field)
+    return [ProtoField(x) for x in self.proto.field]
 
   def GetMessages(self):
     return [ProtoMessage(n, self.qualified_types)
@@ -273,7 +273,7 @@ class ProtoEnum(object):
     return types.TitleCase(self.proto.name)
 
   def Values(self):
-    return map(ProtoEnumValue, self.proto.value)
+    return [ProtoEnumValue(x) for x in self.proto.value]
 
 
 class ProtoEnumValue(object):
